File: algorithms/1_easy/F/03079_find_the_sum_of_encrypted_integers/solution.py
class Solution:
    def sumOfEncryptedInt(self, nums: List[int]) -> int:
        final_res = 0

        for num in nums:
            dig_count = 0
            max_dig = 0

            while num > 0:
                curr_dig = num % 10
                num //= 10

                max_dig = max(max_dig, curr_dig)
                dig_count += 1

            curr_res = 0
            for i in range(dig_count):
                curr_res *= 10
                curr_res += max_dig

            final_res += curr_res

        return final_res

File: algorithms/1_easy/F/03079_find_the_sum_of_encrypted_integers/solution.rs
impl Solution {
    pub fn sum_of_encrypted_int(nums: Vec<i32>) -> i32 {
        let mut final_res = 0_i32;

        for mut num in nums.into_iter() {
            let mut dig_count = 0_i32;
            let mut max_dig = 0_i32;

            while num > 0 {
                let curr_dig = num % 10;
                num /= 10;

                max_dig = max_dig.max(curr_dig);
                dig_count += 1;
            }

            let mut curr_res = 0_i32;
            for i in 0..dig_count {
                curr_res *= 10;
                curr_res += max_dig;
            }

            final_res += curr_res;
        }

        final_res
    }
}
